package FileTransfer;
import "services/file_transfer/checkbook.proto";

message SliceRequest {
  required Slice slice = 1;
  required bytes content = 2;
};

message SliceResponse {
  required bool succeed = 1;
  optional bool finished = 2;
};

message CheckBookResponse {
  required bool succeed = 1;
};

message RegisterRequest {
  required string src_filename = 1;
  required string local_filename = 2;
  required string local_mac_address = 3;
};

message RegisterResponse {
  required bool succeed = 1;
}

service FileTransferService {
  rpc ReceiveCheckBook(CheckBook) returns (CheckBookResponse);
  rpc ReceiveSlice(SliceRequest) returns (SliceResponse);
  rpc Register(RegisterRequest) returns (RegisterResponse);
};
